ch5-catdir/directory.cpp

```diff
@@ -14,7 +14,8 @@ Directory::Directory(const std::string_view& path)
 {
     if (!m_dir)
     {
-        throw std::system_error(errno, std::generic_category(),
+        auto savedErrno = errno;
```

Comment:
`const`?

Reply:
Is it really needed with a variable that is declared in a small block (and used only once)?

Reply:
This is why I said you may decide it doesn't worth changing it :)

Not really "needed", but it's a style I started to try to stick to.